### PR TITLE
Add v4 support

### PR DESCRIPTION
**Pull request by srknzl**
Hi there from current nodejs client maintainer. This pr updates hazelcast client version to v4. v3.12 is becoming obsolete and we are slowly dropping maintenance support. 

Btw if you use delete instead of remove, destroy operation will be faster. You can consider using delete if returned value is not important:

```js
HazelcastStore.prototype.destroy = function (sid, fn) {
    if (!this.client)
      throw new Error('No client found! Call HazelcastStore.setClient(hzInstance)');

    const psid = this.prefix + sid;

    this.sessionsMap.then(map => {
      map.remove(psid).then(value => {
        debug("Item removed. sid:" + sid + ". Previous value: ", value);
        fn(null, sid);
      }).catch(error => {
        debug("Error setting item - " + JSON.stringify(error));
        fn(error);
      });
    });
  };
```

### CODE REVIEW

File: test/test-store.js [hunk 1]
```diff
@@ -31,7 +28,11 @@ describe("hazelcast-store", function () {
       assert(typeof store === "object");
 
       HazelcastClient
-      .newHazelcastClient(clientConfig)
+      .newHazelcastClient({
+        network: {
+          clusterMembers: ['127.0.0.1:5701']
```
Comment (srknzl) [inline test/test-store.js:33]: note: we don't need this config actually since the default value will be enough. added since old code included it

Comment (uclaeamsavino) [reply test/test-store.js:33]: I'll merge and publish this tomorrow. Comment here if I  forget, I'll see the email. 

If you want to change map.remove to map.delete for performance reasons I'm ok with that as long as the tests pass. The code isn't doing anything with the returned value, so it should be fine.

Comment (srknzl) [reply test/test-store.js:33]: yes there is only a debug statement that logs it.

btw readme can also be changed regarding client configuration. I'll add these changes as well.

**Comment by jackspaniel** [comment]
Done. Thanks a lot. If you want to take over this repo let me know. It's not a lot of effort, but I might not be around forever.

**Pull request merged**